### PR TITLE
tests: add univ3 svg generation test

### PR DESCRIPTION
**Pull request by enitrat**
Time spent on this PR: 1d

Adds Mock UniswapV3 contracts that allow querying a dynamically generated on-chain SVG representing a position.
The values are mocked, and due to some issues, the base64 encoding of the image crashes with a stack underflow error (tbd).

However, the encoding of the rest takes ~27M steps

Also: Because `NFTDescriptor` is a library with an `external` function, it must be deployed and called with its address. When referring to this library, the compiler will add a placehold in form `__$HASH_OF_PATH$__` in the compiled bytecode. Thus, I implemented a way of linking a library in the deployments scripts - we can deploy the contract providing a set of required libraries, that are deployed, hashed, and their address replaces the hash in the compiled bytecode.

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves #<Issue number>

## What is the new behavior?

-
-
-

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot/1292)

### CODE REVIEW

**Comment by enitrat** [comment]
good to merge, can ignore the trunk check

File: kakarot_scripts/constants.py [hunk 1]
```diff
@@ -182,6 +182,7 @@ class ChainId(IntEnum):
     16,
 )
 SOURCE_DIR = Path("src")
+SOURCE_PATH = Path("solidity_contracs/src")
```
Comment (ClementWalter) [inline kakarot_scripts/constants.py:185]: naming is confusing and this needs to be taken from the foundry file (see get_solidity_artifacts)

**Review by ClementWalter — CHANGES_REQUESTED** [review]
I'd revert the changes made to `deploy` and use the foundry file to get if there are some

```
    "linkReferences": {
      "solidity_contracts/src/UniswapV3/NFTDescriptor.sol": {
        "NFTDescriptor": [{ "start": 3038, "length": 20 }]
      }
    }
```

and if so, then deploy them first, and replace. Even the `compute_library_identifier` doesn't seem required

File: kakarot_scripts/utils/kakarot.py [hunk 2]
```diff
@@ -140,12 +147,31 @@ def get_contract(
...
+def compute_library_identifier(library_app: str, library_name: str) -> bytes:
+    return keccak(
+        f"{SOURCE_PATH}/{library_app}/{library_name}.sol:{library_name}".encode("utf-8")
+    )[:LIBRARY_PLACEHOLDER_PREFIX_BYTES]
+
+
 async def deploy(
-    contract_app: str, contract_name: str, *args, **kwargs
+    contract: Web3Contract, *args, caller_eoa: Optional[Account] = None, **kwargs
 ) -> Web3Contract:
```
Comment (obatirou) [inline kakarot_scripts/utils/kakarot.py:174]: Deploy is not returning a Web3Contract anymore but `receipt, starknet_address, evm_address`

File: solidity_contracts/src/UniswapV2Router/libraries/UniswapV2Library.sol [hunk 3]
```diff
@@ -33,7 +33,7 @@ library UniswapV2Library {
                         hex"ff",
                         factory,
                         keccak256(abi.encodePacked(token0, token1)),
-                        hex"666a5b78ea0b660c426b08cb5b7427447e909408067de1a5519c772ee9a3c032" // init code hash
+                        hex"0f5b822a8dffa6ce589a2c240d78a6a2b38a51835a97ceab40c1f301e46ba30b" // init code hash
```
Comment (obatirou) [inline solidity_contracts/src/UniswapV2Router/libraries/UniswapV2Library.sol:36]: From where this init code hash was taken ?
Did not find it in commit history of https://github.com/Uniswap/v2-periphery/blob/0335e8f7e1bd1e8d8329fd300aea2ef2f36dd19f/contracts/libraries/UniswapV2Library.sol#L24

Comment (enitrat) [reply solidity_contracts/src/UniswapV2Router/libraries/UniswapV2Library.sol:36]: run `forge build`, open `UniswapV2Library.json` and take the keccak hash of the bytecode hexstring

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**